cloud/migrate-mst-cloud/schema-then-data.md

```diff
@@ -0,0 +1,364 @@
+```
+
+If you have continuous aggregates, you might get the following error. Ignore
+this, because you [restore your aggregates](#recreate-continuous-aggregates)
+later on.
+
+```bash
+pg_restore: error: could not execute query: ERROR:  relation "_timescaledb_internal._materialized_hypertable_x" does not exist
+```
+
+```bash
+​​pg_restore: WARNING:  no privileges were granted for "<..>"
+```
+
+## Restore hypertables in Timescale Cloud
+After pre-data migration, your hypertables from your source database become
+regular PostgreSQL tables in Cloud. Recreate your hypertables in Cloud to
+restore them.
+
+<procedure>
+
+### Restoring hypertables in Timescale Cloud
+1.  Connect to your Timescale Cloud database:
+    ```sql
+    psql "postgres://tsdbadmin:<CLOUD_PASSWORD>@<CLOUD_HOST>:<CLOUD_PORT>/tsdb?sslmode=require"
+    ```
+1.  Restore the hypertable:
+    ```sql
+    SELECT create_hypertable(
+       '<TABLE_NAME>', '<TIME_COLUMN_NAME>',
+        chunk_time_interval =>
+            INTERVAL '<CHUNK_TIME_INTERVAL>');
+    ```
+
+</procedure>
+
+## Copy data from the source database
+After restoring your hypertables, return to your source database to copy your
+data, table by table.
+
+<procedure>
+
+### Copying data from your source database
+1.  Connect to your source database:
+    ```bash
+    psql "postgres://<SOURCE_DB_USERNAME>:<SOURCE_DB_PASSWORD>@<SOURCE_DB_HOST>:<SOURCE_DB_PORT>/<SOURCE_DB_NAME>?sslmode=require"
+    ```
+1.  Dump the data from the first table into a `.csv` file:
+    ```sql
+    \COPY (SELECT * FROM <TABLE_NAME>) TO <TABLE_NAME>.csv CSV
+    ```
+    Repeat for each table and hypertable you want to migrate.
+
+</procedure>
+
+<highlight type="note">
+If your tables are very large, you can migrate each table in multiple pieces.
+Split each table by time range, and copy each range individually. For example:
+
+```sql
+\COPY (SELECT * FROM <TABLE_NAME> WHERE time > ‘2021-11-01’ AND time < ‘2011-11-02’) TO <TABLE_NAME_DATE_RANGE>.csv CSV
+```
+</highlight>
+
+## Restore data into Timescale Cloud
+When you have copied your data into `.csv` files, you can restore it to
+Timescale Cloud by copying from the `.csv` files. There are two methods: using
+regular PostgreSQL [`COPY`][copy], or using the TimescaleDB
+[`timescaledb-parallel-copy`][timescaledb-parallel-copy] function. In tests,
+`timescaledb-parallel-copy` is 16% faster. The `timescaledb-parallel-copy` tool
+is not included by default. You must install the function.
+
+<highlight type="important">
+Both `timescaledb-parallel-copy` and `COPY` decompress compressed data. If you
+provisioned your Timescale Cloud storage for your compressed data, the
+uncompressed data may take too much storage. To avoid this problem, periodically
+recompress your data as you copy it in. For more information on compression, see
+the [compression section](https://docs.timescale.com/timescaledb/latest/how-to-guides/compression/).
```

Comment:
```suggestion
Because `\COPY` decompresses data, any compressed data in your source 
database has been decompressed. If you
provisioned your Timescale Cloud storage for your compressed data, the
uncompressed data may take too much storage. To avoid this problem, periodically
recompress your data as you copy it in. For more information on compression, see
the [compression section](https://docs.timescale.com/timescaledb/latest/how-to-guides/compression/).
```

Reply:
@lukas-bernert if you like this rewrite then I'll add it and we can (finally!) merge the entire thing

Reply:
lgtm